src/studio/ui/StudioPanel.java

```diff
@@ -741,8 +796,14 @@ else if ((row == 1) && ((noWins % 3) > 1))
         }
     }
 
-    private void setServer(Server server) {
-        if (server == null) return;
+    private void setServer(Server server, boolean addToNavigationHistory) {
+        if (server == null)
+            return;
+
+        if (addToNavigationHistory)
+            serverNavigateHistory.addSelectedItem(server);
+        prevServerAction.setEnabled(serverNavigateHistory.hasPreviousItem());
+        nextServerAction.setEnabled(serverNavigateHistory.hasNextItem());
```

Comment:
And below (hidden in the review) is the reason I don't like the concept of NavigationHistory (which I already advertised earlier). We simply already have serverHistory. I think we need one navigation or serverHistory. The logic should be combined.

Reply:
... and serverHistoryAction should move a selectedIndex pointer in the navigationHistory, rather than adding an item in the end of the navigation